populus/compilation/backends/solc_auto.py

```diff
@@ -16,37 +16,30 @@
 from .base import (
     BaseCompilerBackend,
 )
-from .solc_combined_json import (
-    SolcCombinedJSONBackend,
-)
+
 from .solc_standard_json import (
     SolcStandardJSONBackend,
 )
 
 
-def get_solc_backend_class_for_version(solc_version):
-    if is_string(solc_version):
+def get_solc_backend_class_for_version(solc_version=None,settings=None):
+
+    if solc_version == None:
+        solc_version = get_solc_version()
+
+    elif is_string(solc_version):
         solc_version = Version(solc_version)
 
-    if solc_version in Spec('<=0.4.8'):
-        return SolcCombinedJSONBackend
-    elif solc_version in Spec('>=0.4.11'):
-        return SolcStandardJSONBackend
+    if solc_version in Spec('>=0.4.11'):
+        backend_class = SolcStandardJSONBackend
     else:
         raise OSError(
-            "The installed solc compiler is not supported.  Supported versions "
-            "of the solc compiler are <=0.4.8 and >=0.4.11"
+            "The installed solc compiler is not supported. Supported versions "
+            "solc >=0.4.11"
         )
 
+    if settings == None:
+        return backend_class()
+    else:
+        return backend_class(settings)
```

Comment:
Hrm, rather than having settings be passed in as an argument like this, we could instead pass it in as `**settings` which would allow removal of this `if/else` clause.